### ISSUE TITLE
'DeploymentOutputs' variable doesn't contain the ids of the triggered pipelines

### ISSUE THREAD

**Issue by Camiloasc1** [liprec/vsts-publish-adf#70]
Hello,

I'm using the task to trigger a pipeline but then I'm not able to read the output run ids as indicated in the documentation.
![image](https://user-images.githubusercontent.com/6014221/116762481-6d2e3000-a9cf-11eb-9362-09633bc6c3a2.png)

As you can see below, I'm trying to use `AdfPipelineResult` as the variable where I want the run ids. The pipeline is successfully triggered but then the value I read from the variable is an empty json array `[]`.

I'm suspecting it is related to [this line](https://github.com/liprec/vsts-publish-adf/blob/26a83b3f512508a2dffbca707dd2237665e8afaf/trigger-adf-pipeline/v2/triggeradfpipeline.ts#L269) where the filter seems to be looking for failed results instead of the successful ones to be passed on to the json serializer. Is this correct?
`JSON.stringify(results.filter((result: DataFactoryRunResult) => !result))`

### Azure Data Factory Trigger Pipeline
```
steps:
- task: liprec.vsts-publish-adf.trigger-adf-pipeline.trigger-adf-pipeline@2
  displayName: 'Trigger pipeline run MyPipeline'
  inputs:
    azureSubscription: '***'
    ResourceGroupName: '***'
    DatafactoryName: '***'
    PipelineFilter: MyPipeline
    DeploymentOutputs: AdfPipelineResult
```

### My PowerShell script:
```
Write-Host "Value for the 'DeploymentOutputs' variable:"
Write-Host "$(AdfPipelineResult)"
```

### Azure Data Factory Trigger Pipeline
```
2021-04-30T23:14:30.0868632Z ##[section]Starting: Trigger pipeline run MyPipeline
2021-04-30T23:14:30.1073692Z ==============================================================================
2021-04-30T23:14:30.1074085Z Task         : Azure Data Factory Trigger Pipeline
2021-04-30T23:14:30.1074417Z Description  : Trigger Azure Data Factory V2 Pipelines
2021-04-30T23:14:30.1074665Z Version      : 2.3.3
2021-04-30T23:14:30.1074918Z Author       : Jan Pieter Posthuma
2021-04-30T23:14:30.1076016Z Help         : [More Information](https://github.com/liprec/vsts-publish-adf)
2021-04-30T23:14:30.1076445Z ==============================================================================
2021-04-30T23:14:33.8834281Z Found 1 pipeline(s).
2021-04-30T23:14:34.6215165Z Pipeline 'MyPipeline' triggered with run id: 'f430e22a-cb00-425c-8eda-716edb7484b2'.
2021-04-30T23:14:34.6254059Z Updated output variable 'AdfPipelineResult', which contains the RUN IDs of the package triggers in string format.
2021-04-30T23:14:34.6361545Z ##[section]Finishing: Trigger pipeline run MyPipeline
```

### My PowerShell script:
```
2021-04-30T23:14:34.6414302Z ##[section]Starting: Wait for completion
2021-04-30T23:14:34.6550960Z ==============================================================================
2021-04-30T23:14:34.6551300Z Task         : PowerShell
2021-04-30T23:14:34.6551571Z Description  : Run a PowerShell script on Linux, macOS, or Windows
2021-04-30T23:14:34.6552196Z Version      : 2.180.1
2021-04-30T23:14:34.6552428Z Author       : Microsoft Corporation
2021-04-30T23:14:34.6552791Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
2021-04-30T23:14:34.6553192Z ==============================================================================
2021-04-30T23:14:35.7059549Z Generating script.
2021-04-30T23:14:35.7535045Z ========================== Starting Command Output ===========================
2021-04-30T23:14:35.7812151Z ##[command]"C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'D:\a\_temp\1b03baa3-c28d-482b-909d-2378f13e2a82.ps1'"
2021-04-30T23:14:36.0723629Z Value for the 'DeploymentOutputs' variable:
2021-04-30T23:14:36.0730635Z []
2021-04-30T23:14:36.1736504Z ##[section]Finishing: Wait for completion
```

Any help would be greatly appreciated.

Thanks.

**Comment by liprec** [comment]
Thanks for the feedback and I will take a look what is going wrong

**Comment by andb0t** [comment]
Hi everyone,
any news on this issue? This would greatly speed up my deployment pipeline. My use case is to trigger and then wait until the pipeline is finished. For this I need to query the pipeline status using the ID. Unfortunately, it's empty for now `[]` !

**Comment by liprec** [comment]
Sorry, but I still need to push a new version with this fix. Hopefully I have the time anytime soon.

**Comment by andb0t** [comment]
I keep my fingers crossed! Thanks for answering!

**Comment by liprec** [comment]
Fixed in v2.4.0
